src/metabase/serialization/serialize.clj

```diff
@@ -0,0 +1,141 @@
+(ns metabase.serialization.serialize
+  "Transform entity into a form suitable for serialization."
+  (:require [clojure.string :as str]
+            [medley.core :as m]
+            [metabase.mbql
+             [normalize :as mbql.normalize]
+             [util :as mbql.util]]
+            [metabase.models
+             [card :refer [Card]]
+             [dashboard :refer [Dashboard]]
+             [dashboard-card :refer [DashboardCard]]
+             [dashboard-card-series :refer [DashboardCardSeries]]
+             [database :refer [Database] :as database]
+             [field :refer [Field] :as field]
+             [metric :refer [Metric]]
+             [pulse :refer [Pulse]]
+             [pulse-card :refer [PulseCard]]
+             [pulse-channel :refer [PulseChannel]]
+             [segment :refer [Segment]]
+             [table :refer [Table]]
+             [user :refer [User]]]
+            [metabase.serialization.names :refer [fully-qualified-name]]
+            [metabase.util :as u]
+            [toucan.db :as db]))
+
+(def ^:private ^{:arglists '([form])} mbql-entity-reference?
+  "Is given form an MBQL entity reference?"
+  (partial mbql.normalize/is-clause? #{:field-id :fk-> :metric :segment}))
+
+(defn- humanize-mbql
+  [mbql]
+  (-> mbql
+      mbql.normalize/normalize
+      (mbql.util/replace
+        ;; `integer?` guard is here to make the operation idempotent
+        [:field-id (id :guard integer?)]
+        [:field-id (fully-qualified-name Field id)]
+
+        [:metric (id :guard integer?)]
+        [:metric (fully-qualified-name Metric id)]
+
+        [:segment (id :guard integer?)]
+        [:segment (fully-qualified-name Segment id)]
+
+        ;; Legacy form with raw IDs
+        [:fk-> (from :guard integer?) (to :guard integer?)]
+        [:fk-> [:field-id (fully-qualified-name Field from)]
+         [:field-id (fully-qualified-name Field to)]])))
+
+(defn- humanize-entity-references
```

Comment:
Instead of naming these functions/methods with the `humanize-` prefix, how about `serialize-`? I think that conveys what's going on here a bit better than the term `humanize` which we're already using for the Table/Field name humanization stuff and the connection error message humanization stuff. When I see this my mind jumps to "make ugly strings look nicer for user consumption", not "replace integer IDs with literal names" 

Idk. Just my 2 cents. Don't feel super strongly about this